library/std/src/lib.rs

```diff
@@ -212,6 +212,7 @@
     all(target_vendor = "fortanix", target_env = "sgx"),
     feature(slice_index_methods, coerce_unsized, sgx_platform)
 )]
+#![deny(rustc::existing_doc_keyword)]
```

Comment:
```suggestion
#![warn(rustc::existing_doc_keyword)]
```
std is compiled with `-Dwarnings` by default so I think that using `warn` here is better as it allows for people to disable without having to worry about any of this :sweat_smile:

Reply:
Well, since it's very specific thing, if anything is invalid, it has to throw an error right away. So I think it's better to use `deny` here.

Reply:
> it has to throw an error right away.

won't it just document a non-existing keyword if something is invalid? I think using `warn` has the small advantage of not failing compilation in the extremely rare case that someone removes an unstable keyword and wants to do some other stuff with `deny_warnings = false` as well as being more consistent with other lints used inside of std.

But this really doesn't matter enough for me to block this PR :laughing:

Reply:
Sorry but it was a hard error and this check is supposed to be a hard one too. :laughing:

Reply:
ah, so if it's intended to be a hard error why are you using a lint here :sweat_smile:

Reply:
Because it's supposed to be a hard error **only** for std. Explanations: in the `std` we only want to use it for existing keywords. If the keyword documented doesn't exist, it means it's a typo. Outside of the std, proc-macro crates owners might add new keywords and document them. In these cases, we don't want to warn or anything. (Yep, that's quite the thing XD)

Reply:
ah, ok that makes sense :laughing: